### ISSUE TITLE
What is with Python 3.6 support?

### ISSUE THREAD

**Issue by hrw** [tensorflow/io#1598]
I got tensorflow-io built with Python 3.6.15:

```
tensorflow_io-0.23.1-cp36-cp36m-manylinux_2_17_aarch64.manylinux2014_aarch64.whl
tensorflow_io_gcs_filesystem-0.23.1-cp36-cp36m-manylinux_2_17_aarch64.manylinux2014_aarch64.whl
```

But they are not installable:

```
TASK [tensorflow-io : install Python package] **********************************
Friday 17 December 2021  12:46:37 +0000 (0:00:00.066)       0:26:25.420 *******
fatal: [localhost]: FAILED! => changed=false
  cmd:
  - /tmp/workspace/venv-cp36-cp36m/bin/pip
  - install
  - --extra-index-url
  - https://snapshots.linaro.org/ldcg/python-cache/
  - /tmp/workspace/io-0.23.1/dist/tensorflow_io_gcs_filesystem-0.23.1-cp36-cp36m-linux_aarch64.whl
  msg: |-
    stdout: Looking in indexes: https://pypi.org/simple, https://snapshots.linaro.org/ldcg/python-cache/
    Processing ./workspace/io-0.23.1/dist/tensorflow_io_gcs_filesystem-0.23.1-cp36-cp36m-linux_aarch64.whl

    :stderr: ERROR: Package 'tensorflow-io-gcs-filesystem' requires a different Python: 3.6.15 not in '<3.11,>=3.7'```

Looks like I missed point when Python 3.6 support got removed from both Tensorflow and Tensorflow IO.

**Comment by yongtang** [comment]
@hrw Tensorflow 2.7 has dropped 3.6 support (https://pypi.org/project/tensorflow/2.7.0/#description), as python 3.6 is EOL'ed this year https://endoflife.date/python

Since tensorflow-io depends on TF, we have dropped python 3.6 support for 0.23.0 as well.

A little more on python 3.7: Numpy has decided to drop 3.7 support in 1.22.0 (https://github.com/numpy/numpy/releases_) even though python 3.7's EOL is 1.5 years ago.
TF depends on numpy and that is problematic.

In our latest SIG IO monthly meeting we discussed with TF and so far the conclusion is that TF will support 3.7 (but lower numpy dependency for 3.7). So 3.7 will not be deprecated for TF and tensorflow-io.
